src/cdk/constructs/auto-delete-bucket/auto-delete-bucket.test.ts

```diff
@@ -0,0 +1,54 @@
+import '@aws-cdk/assert/jest'
+import { Stack, RemovalPolicy } from '@aws-cdk/core'
+
+import { AutoDeleteBucket } from '.'
+
+describe('Auto-delete bucket', () => {
+  it('throws an error if wrong removalPolicy is passed as a prop', () => {
+    const stack = new Stack()
+
+    expect(() => {
+      new AutoDeleteBucket(stack, 'ADB', {
+        removalPolicy: RemovalPolicy.RETAIN
+      })
+    }).toThrowError('"removalPolicy" must be DESTROY')
+  })
+
+  it('creates the bucket and all necesarry resources', () => {
+    const stack = new Stack()
+    new AutoDeleteBucket(stack, 'ADB')
+
+    expect(stack).toHaveResource('AWS::S3::Bucket')
+
+    expect(stack).toHaveResourceLike('AWS::Lambda::Function', {
+      Handler: 'lambda/index.handler',
+      Runtime: 'nodejs10.x',
+      Timeout: 900
+    })
+
+    expect(stack).toHaveResource('AWS::IAM::Policy')
+    //   expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
```

Comment:
can we remove code if it's not used?

Reply:
If it's ok with you, I'd rather keep it as I really want to have it in at some point (i'll keep an eye on it, I promise).
It highlights that there should be a better test there but the lack of support from @aws-cdk/assert stops us from doing that at this point.

Reply:
Can you add simple todo near this comment? ;)